src/Compilers/Core/MSBuildTask/Shared/ManagedCompiler.cs

```diff
@@ -689,6 +687,14 @@ protected internal virtual void AddResponseFileCommands(CommandLineBuilderExtens
             commandLine.AppendFileNamesIfNotNull(Sources, " ");
         }
 
+        internal void AddResponseFileCommandsForSwitchesSinceInitialReleaseThatAreNeededByTheHost(CommandLineBuilderExtension commandLine)
```

Comment:
While `AddResponseFileCommandsForSwitchesSinceInitialReleaseThatAreNeededByTheHost` is very descriptive could it just be `AddNeededSwitchesToResponseFile` and have a doc comment explaining what this method does?